### ISSUE TITLE
[Bug] Memory leak caused by fetch proxy

### ISSUE THREAD

**Issue by Yidadaa** [Yidadaa/ChatGPT-Next-Web#1598]
https://github.com/Yidadaa/ChatGPT-Next-Web/blob/38bffd423c7e2fd0210ecb768a33785c97ab2961/app/api/common.ts#L28-L38

We should use `AbortController` to destroy this `fetch` response after 600 seconds, or it will cause memory leak.

![8aa5edcf4abcf6724a20dc343578518e](https://github.com/Yidadaa/ChatGPT-Next-Web/assets/16968934/a9a8feb9-457c-4bd0-89b7-a5df06a1e57e)

**Comment by Yidadaa** [comment]
Here are some hints:

```ts
// setup AbortController
const controller = new AbortController();
// signal to pass to fetch
const signal = controller.signal;

setTimeout(() => controller.abort(), 60 * 1000);

// pass signal to fetch init options
```

ref: https://stackoverflow.com/questions/31061838/how-do-i-cancel-an-http-fetch-request

**Comment by Yidadaa** [comment]
fixed by #1604
